### DISCUSSION TITLE
undefined error in x-for after upgrading to v3

### DISCUSSION THREAD

**Discussion by JonathanLouw** [livewire/livewire#7475] [7. Bugs (archived)]
### Livewire version

3.2.6

### Which PHP version are you using?

PHP 8.1

### Repository URL

https://github.com/JonathanLouw/livewire-playground

### Steps To Reproduce

clone the provided repo and navigate to /example route or set up your own deferred loading example page:

```
class ExampleComponent extends Component
{
    public bool $load = false;

    public function loadUsers(): void
    {
        $this->load = true;
    }

    public function render()
    {
        return view('livewire.example-component', [
            'users' => $this->load
                ? User::query()->inRandomOrder()->limit(5)->get()
                : collect(),
        ]);
    }
}
```
```
<div>
    <h1>Example Component</h1>
    <button wire:click.prevent="loadUsers">Load</button>
    <div x-data="{ users: @js($users) }">
        <template x-for="user in users">
            <div x-text="user.name"></div>
        </template>
    </div>
</div>
```
```
Route::get('/example', \App\Livewire\ExampleComponent::class);
```
Then try to click load to fetch the results

### Additional information

With the provided example the behaviour I'm getting is this:

click "load" button once, nothing happens, click it a second time and I get a warning that user is undefined, for example:

```
Alpine Expression Error: user is not defined

Expression: "user.name"
```
it also output a console error pointing to this bit of code:
```
(async function anonymous(__self,scope
) {
with (scope) { __self.result = user.name }; __self.finished = true; return __self.result;
})
```
And strangely enough when I click it a third time it starts working perfectly fine and correctly fetches 5 random users on every click.

In various other places we are using alpine in our project we're getting the same alpine errors like the one above where a lot of errors are being output that things are undefined which all worked fine on v2 but not after upgrading to v3.

### Please confirm (incomplete submissions will not be addressed)

- [X] I have provided easy and step-by-step instructions to reproduce the bug.
- [X] I have provided code samples as text and NOT images.
- [X] I understand my bug report will be removed if I haven't met the criteria above.

**Comment by gofish543** [comment]
I am having this issue all over the place. I literally cannot get a select dropdown to work without a resolution to this issue

**Reply by gofish543** [reply]
Bumping this. This is a serious problem

**Reply by JonathanLouw** [reply]
Hey I'm still having this issue as well however I've managed to come up with a few potential work arounds for now that might help you or someone else stumbling upon this until this anonymous scope bug is fixed:
1. With regards to the deferred loading logic above i've decided to instead use a method as an ajax request e.g. `$wire.getUsers().then(users => this.users = users)`
2. Try use `this.` when referencing alpine state as i noticed this fixed it in a few places
3. If you're still using laravel mix and are using custom alpine magic methods, stores, plugins etc. you will need to change to vite as it's not going to work with mix if you [follow these instructions](https://livewire.laravel.com/docs/installation#manually-bundling-livewire-and-alpine).

**Comment by gofish543** [comment]
@joshhanley Do you have any idea to what is causing this? It is and has been holding back a new application deploy for the past two months now.

**Reply by joshhanley** [reply]
@gofish543 No idea sorry. Have you tried what @JonathanLouw suggested above? Otherwise best thing to do is submit a failing test PR so it can be investigated.

**Reply by gofish543** [reply]
Of the three offered work-around the first does not work for the use case specified and the other two are not fixes but recommendations

**Comment by EdgarsJoja** [comment]
I encountered the same issue (also for the same/similar use case - searchable select). Because the select options have to be dynamic (and doing livewire request to server just for filtering list is an overkill), I'm rendering them with Alpine. On the initial page load it works. However when state changes (and available options change as well), after component re-render this error appears.

This discussion in Alpine repo seems to be the same: https://github.com/alpinejs/alpine/discussions/1383
